gamemodes/horde/gamemode/cl_init.lua

```diff
@@ -338,8 +338,9 @@ net.Receive("Horde_ForceCloseShop", function ()
     end
 
     HORDE.TipPanel:SetVisible(false)
-    HORDE.leader_board:SetVisible(false)
-
+    if LocalPlayer():GetInfoNum("horde_show_leaderboard", 0) == 0 then   
```

Comment:
don't use GetInfoNum for clientside convars, this should be entirely clientside
Use a localized GetConVar and :GetBool() it